web/docs/language/features.md

```diff
@@ -178,138 +178,247 @@ Currently entities can be accessed only in Operations (Queries & Actions), so ch
+To create a Wasp Query, you must:
+1. Define the Query's NodeJS implementation
+2. Declare the Query in Wasp using the `query` declaration
+
+After completing these two steps, you'll be able to use the Query from any point in your code. 
+
+
+#### Defining the Query's NodeJS implementation
+A Query must be implemented as an `async` NodeJS function that takes two arguments.
+Since both arguments are positional, you can name the parameters however you want, but we'll stick with `args` and `context`:
+1. `args`:  An object containing all the arguments (i.e., payload) sent to the Query (e.g., filtering conditions).
+Take a look at [the examples of usage](#using-the-query) to see how to pass this object to the Query.
+3. `context`: An additional context object. This object contains user session information, as well as information about entities. The examples here won't use the context for simplicity purposes. You can read more about it in the [section about using entities in queries](#using-entities-in-queries).
+
+Here's an example of two simple Queries:
+```js title="ext/queries.js"
+// our "database"
+const tasks = [
+    { id: 1, description: "Buy some eggs", isDone: true },
+    { id: 2, description: "Make an omelette", isDone: false },
+    { id: 3, description: "Eat breakfast", isDone: false }
+]
+
+
+// You don't need to use the arguments if you don't need them
+export const getAllTasks = async () => {
+  return tasks;
+}
+
+// The 'args' object is something sent by the caller (most often from the client) 
+export const getFilteredTasks = async (args) => {
+  const { isDone } = args;
+  return tasks.filter(task => task.isDone === isDone)
 }
 ```
-`query` declaration type has two fields:
-- `fn: ExtImport` (required)
-- `entities: [Entity]` (optional)
 
-2. Implemenation in NodeJS:
-```js title="ext/foo.js"
+#### Declaring a Query in Wasp 
+After implementing your Queries in NodeJS, all that's left to do before using them is tell Wasp about it!
+You can easily do this with the `query` declaration, which supports the following fields:
+- `fn: ExtImport` (required) - The import statement of the Query's NodeJs implementation 
+- `entities: [Entity]` (optional) - A list of entities you wish to use inside your Query.
+We'll leave this option aside for now. You can read more about it [here](#using-entities-in-queries).
+
+Wasp Queries and their implementations don't need to (but can) have the same name, so we will keep the names different to avoid confusion.
+With that in mind, this is how you might declare the Queries that use the implementations from the previous step:
+```c title="main.wasp"
 // ...
-export getAllTasks = async (args, context) => {
-  return [
-    { description: "Buy some eggs", isDone: true },
-    { description: "Make an omelette", isDone: false }
-  ]
+
+query fetchAllTasks {
+  fn: import { getAllTasks } from "@ext/queries.js"
+}
+
+query fetchFilteredTasks {
+  fn: import { getFilteredTasks } from "@ext/queries.js"
```

Comment:
I changed the `query` names here to emphasize the difference between a Wasp query and its NodeJS implementation.

Reply:
Aha, got it!
It is good to demonstrate this, but it sucks a bit that names are not consistent, as there is no good reason to call some of them fetch and some of them get. But ok, I guess it is ok for the purposes of an example!

Reply:
Yes, the names are definitely arbitrary.

I couldn't come up with any other way to make it clear that what we're using are the things declared in Wasp. I also use these terms later in the text when explaining the differences between the query (`fetchFilteredTasks`) and its implementation (`getFilteredTasks`). We were using the term query interchangeably for these two concepts, and I thought it could cause confusion.

I can change the names to be the same again if you want. However, an even better option would be to keep them different but meaningful (ideas welcome :)).

Reply:
Great exchange here, this definitely reflects what I also experienced while developing Wasp apps. Usually, it is the same name, but we currently can't avoid stating it twice. As an only additional thing I can think of we could do right now is add a comment above query - e.g. "this doesn't have to be the same name, although in practice it often will be."

Reply:
We already mention that in the text, but I can add a comment too, no problem.